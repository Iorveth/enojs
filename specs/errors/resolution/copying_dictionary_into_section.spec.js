const eno = require('../../../eno.js');

const input = `
original:
entry = value

# copy < original
`.trim();

describe('resolution.copyingDictionaryIntoSection', () => {

  let error;

  beforeAll(() => {
    try {
      eno.parse(input);
    } catch(err) {
      error = err;
    }
  })

  it(`provides the correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toMatchSnapshot();
  });
});
