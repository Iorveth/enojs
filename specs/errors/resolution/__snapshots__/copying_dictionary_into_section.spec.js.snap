// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolution.copyingDictionaryIntoSection provides correct selection metadata 1`] = `
Array [
  Array [
    3,
    0,
  ],
  Array [
    3,
    17,
  ],
]
`;

exports[`resolution.copyingDictionaryIntoSection provides the correct message 1`] = `
"In line 4 a dictionary is copied into a section.

   Line | Content
   ...
      2 | entry = value
      3 | 
 >    4 | # copy < original
"
`;
